lib/arn.go

```diff
@@ -57,3 +59,7 @@ func (a *ARN) String() string {
 
 	return strings.Join(components, ":")
 }
+
+func (a *ARN) ValidRole() bool {
```

Comment:
This is vestigial at this point (we were using it in a previous version of this pull request), but since we're not using any more, is there a reason to keep it? (I know, I hate throwing away working code too… 😝, but at this point it's just adding maintenance complexity without needing to be here).